dune_client/api/base.py

```diff
@@ -0,0 +1,115 @@
+""""
+Basic Dune Client Class responsible for refreshing Dune Queries
+Framework built on Dune's API Documentation
+https://duneanalytics.notion.site/API-Documentation-1b93d16e0fa941398e15047f643e003a
+"""
+from __future__ import annotations
+
+import logging.config
+import os
+from json import JSONDecodeError
+from typing import Dict, Optional, Any
+
+import requests
+from requests import Response
+
+
+# pylint: disable=too-few-public-methods
+class BaseDuneClient:
+    """
+    A Base Client for Dune which sets up default values
+    and provides some convenient functions to use in other clients
+    """
+
+    BASE_URL = "https://api.dune.com"
+    DEFAULT_TIMEOUT = 10
```

Comment:
Yea definitely -- although I think that is a task for another PR (since this is just refactoring current functionality).

Reply:
Made an issue; https://github.com/bh2smith/dune-client/issues/82

Not sure why BASE_URL would be configurable (at least not at the moment).

Reply:
> Not sure why BASE_URL would be configurable (at least not at the moment).

Useful if we want to use it to run against a staging environment for example.

Reply:
Right! Actually I was thinking it might make even more sense to use staging for CI here. Then our tests wouldn't be affecting production (i.e. editing and creating new queries etc..)

Reply:
Done here: https://github.com/duneanalytics/dune-client/pull/88